AmplifyPlugins/DataStore/Tests/AWSDataStorePluginTests/Sync/Support/MutationEventExtensionsTests.swift

```diff
@@ -0,0 +1,404 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import SQLite
+import XCTest
+
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSDataStorePlugin
+@testable import AWSPluginsCore
+
+// TODO: This flaky test has been disabled, tracking issue: https://github.com/aws-amplify/amplify-ios/issues/1831
+// swiftlint:disable type_body_length
+class MutationEventExtensionsTest: BaseDataStoreTests {
+
+    /// - Given: A pending mutation events queue with event containing `nil` version, a sent mutation
+    ///         event model that matches the received mutation sync model. The received mutation sync has version 1.
+    /// - When: The sent model matches the received model and the first pending mutation event version is `nil`.
+    /// - Then: The pending mutation event version should be updated to the received model version of 1.
+    func testSentModelWithNilVersion_Reconciled() throws {
+        throw XCTSkip("TODO: fix this test")
```

Comment:
The issue https://github.com/aws-amplify/amplify-swift/issues/1831 is pretty old and talks about timing issues related to acquiring the locks. I think there were some changes to how our lock was implemented. Could you remove the XCTSkip and try the test?

Reply:
will give a try